packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -7,26 +12,231 @@ import { IVirtualService } from './virtual-service';
+   * When `false`, retains the original hostname/prefix from the request.
+   *
+   * @default true
+   */
+  readonly rewriteRequestHostname?: boolean;
+
+  /**
+   * The in-coming request's path to be rewritten when the request is matched.
+   *
+   * @default - matched prefix is rewritten to '/'.
+   */
+  readonly pathRewrite?: HttpGatewayRoutePathRewrite;
 }
 
 /**
  * The criterion for determining a request match for this GatewayRoute
  */
 export interface GrpcGatewayRouteMatch {
   /**
-   * The fully qualified domain name for the service to match from the request
+   * Create service name based gRPC gateway route match.
+   *
+   * @default - no matching on service name.
+   */
+  readonly serviceName?: string;
+
+  /**
+   * Create host name based gRPC gateway route match.
+   *
+   * @default - No matching on host name.
+   */
+  readonly hostname?: GatewayRouteHostnameMatch;
+
+  /**
+   * Create metadata based gRPC gateway route match.
+   * All specified metadata must match for the route to match.
+   *
+   * @default - not matching on metadata.
+   */
+  readonly metadata?: HeaderMatch[];
+
+  /**
+   * When `true`, rewrites the original request received at the Virtual Gateway to the destination Virtual Service name.
+   * When `false`, retains the original hostname/prefix from the request.
+   *
+   * @default true
+   */
+  readonly rewriteRequestHostname?: boolean;
+}
+
+/**
+ * Path and prefix properties for HTTP gateway route rewrite.
+ */
+export interface HttpGatewayRoutePathRewriteConfig {
+  /**
+   * GatewayRoute CFN configuration for HTTP gateway route path rewrite.
+   *
+   * @default - rewrite path to '/'.
+   */
+  readonly path?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty;
+
+  /**
+   * GatewayRoute CFN configuration for HTTP gateway route prefix rewrite.
+   *
+   * @default - rewrite prefix to '/'.
+   */
+  readonly prefix?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Used to generate HTTP gateway route rewrite other than the host name.
+ */
+export abstract class HttpGatewayRoutePathRewrite {
+  /**
+   * The path to rewrite.
+   *
+   * @param exact The exact path to rewrite.
+   */
+  public static exact(exact: string): HttpGatewayRoutePathRewrite {
+    return new HttpGatewayRoutePathRewriteImpl(exact);
+  }
+
+  /**
+   * The default prefix used to replace the incoming route prefix when rewritten.
+   * When enabled, rewrites the matched prefix in Gateway Route to '/'.
+   * When disabled, retains the original prefix from the request.
+   */
+  public static disableDefaultPrefix(): HttpGatewayRoutePathRewrite {
+    return new HttpGatewayRoutePrefixRewriteImpl({ defaultPrefix: 'DISABLED' });
+  }
+
+  /**
+   * Replace the incoming route prefix when rewritten.
+   *
+   * @param value The value used to replace the incoming route prefix when rewritten.
+   */
```

Comment:
I'm sorry, I still don't understand this method. How is this different than `exact`?

Reply:
Please refer [this response](https://github.com/aws/aws-cdk/pull/15395#discussion_r665627417).